nmtwrapper.go

```diff
@@ -0,0 +1,142 @@
+package rsmt2d
+
+// The contents of this file have been adapted from the source file available at https://github.com/celestiaorg/celestia-app/blob/main/pkg/wrapper/nmt_wrapper.go,
```

Comment:
the only thing the wrapper imports is the [consts](https://github.com/celestiaorg/celestia-app/blob/5bcf4ad51756e5f31dd7d9ea140844063b6ce391/pkg/wrapper/nmt_wrapper.go#L6C53-L6C53), so maybe it makes sense to accept those consts as args, then move the entire wrapper here

Reply:
Thanks for the suggestion, here is the tracking issue https://github.com/celestiaorg/rsmt2d/issues/248.

Reply:
That also makes sense to me, cannot see any immediate issue for doing so, however, since it requires some refactoring in the code, I'd say it merits a separate PR. Please see the tracking issue https://github.com/celestiaorg/rsmt2d/issues/248.